HPCC-17053 Change in Dropzone Config

g-pan opened the pull request:
Fix HPCC-17053 Change in Dropzone Config
Document the changes in the Dropzone configuration
in the Using the Configmgr doc.
Note: also added logic to automatically generate the attribute table from the XSD file. 

@JamesDeFabia please review
@rpastrana please review


Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

g-pan committed:
HPCC-17053 Change in Dropzone Config

Fix HPCC-17053 Change in Dropzone Config
Document the changes in the Dropzone configuration
in the Using the Configmgr doc.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

JamesDeFabia commented:
Looks good from my POV

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: ac694b5b958e8fba90b7e28fc2b0dddc4eaccddd
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 93.538 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |726 |726 |0 |
|test (thor) |619 |619 |0 |
|test (roxie) |747 |747 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

g-pan committed:
HPCC-17053 Change in Dropzone Config

Fix HPCC-17053 Change in Dropzone Config
Document the changes in the Dropzone configuration
in the Using the Configmgr doc.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

rpastrana commented on docs/UsingConfigManager/UsingConfigManager.xml:
```diff
@@ -1119,19 +1121,35 @@ sudo -u hpcc cp /etc/HPCCSystems/source/NewEnvironment.xml /etc/HPCCSystems/envi
           <para>This tab allows you to add any servers that you wish to
-          configure as drop zones. To add a server to the available drop
-          zones. <orderedlist>
+          configure as a part of this drop zone. To add a server to the
```
the last sentence seems incomplete, you might want to terminate it with a ":" and maybe place it in its own line.

JamesDeFabia replied:
I think he meant to end with a colon.
"To add a server to the current drop zone:" 
However, the doc doesn't clarify the difference between adding more servers to one drop zone component and adding additional drop zone components.

rpastrana commented on docs/UsingConfigManager/UsingConfigManager.xml:
```diff
@@ -1119,19 +1121,35 @@ sudo -u hpcc cp /etc/HPCCSystems/source/NewEnvironment.xml /etc/HPCCSystems/envi
 
               <listitem>
-                <para>Double-click on the Server Address field.</para>
+                <para>RIght-click on the Server Address field and choose
```
there's no need to capitalize the I in RIght

rpastrana reviewed (changes requested):
@g-pan a couple of comments

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 23a77f3950f11cba78259ab9401d42544608c554
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 93.085 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |728 |728 |0 |
|test (thor) |622 |622 |0 |
|test (roxie) |749 |749 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

g-pan committed:
HPCC-17053 Change in Dropzone Config

Fix HPCC-17053 Change in Dropzone Config
Document the changes in the Dropzone configuration
in the Using the Configmgr doc.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

g-pan commented:
Made revisions, new doc can be reviewed at : 
http://10.240.32.243/view/Docs/job/DocDev05/2/CE=ubuntu-14.04-amd64-1/artifact/build-docs/Release/docs/UsingConfigManager-6.3.0-trunk0.pdf
or download the UsingConfigManager file from:
http://10.240.32.243/view/Docs/job/DocDev05/2/CE=ubuntu-14.04-amd64-1/

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 95d3b389749468e4aa87f7a4479f42048c5c8d76
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 93.59 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |728 |728 |0 |
|test (thor) |622 |622 |0 |
|test (roxie) |749 |749 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

g-pan committed:
HPCC-17053 Change in Dropzone Config

Fix HPCC-17053 Change in Dropzone Config
Document the changes in the Dropzone configuration
in the Using the Configmgr doc.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 43b6031ade81e482260cf5a2c6f0cb4295e08d73
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 96.172 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |728 |728 |0 |
|test (thor) |622 |622 |0 |
|test (roxie) |749 |749 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

g-pan committed:
HPCC-17053 Change in Dropzone Config

Fix HPCC-17053 Change in Dropzone Config
Document the changes in the Dropzone configuration
in the Using the Configmgr doc.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: c40444deead20a6428b7fe04ac60b3a24866d1a1
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 95.329 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |728 |728 |0 |
|test (thor) |622 |622 |0 |
|test (roxie) |749 |749 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

g-pan committed:
HPCC-17053 Change in Dropzone Config

Fix HPCC-17053 Change in Dropzone Config
Document the changes in the Dropzone configuration
in the Using the Configmgr doc.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 1638147e2d264ae7a8f9998c9490162792452fa1
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 93.16 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |728 |728 |0 |
|test (thor) |622 |622 |0 |
|test (roxie) |749 |749 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
@rpastrana Back to you

g-pan committed:
HPCC-17053 Change in Dropzone Config

Fix HPCC-17053 Change in Dropzone Config
Document the changes in the Dropzone configuration
in the Using the Configmgr doc.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

rpastrana commented:
holding off on review per @g-pan

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: b0916fe1e37bfd4345f121b96a0d4110262349ea
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.989 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |728 |728 |0 |
|test (thor) |622 |622 |0 |
|test (roxie) |749 |749 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

rpastrana commented:
@g-pan PDF looks reasonable
